lxd/resources/system.go

```diff
@@ -26,7 +30,14 @@ func GetSystem() (*api.ResourcesSystem, error) {
 	system.Type = systemType
 
 	if !sysfsExists(sysClassDMIID) {
-		return &system, nil
+		system, err := getSystemFromLshw()
+		if err != nil {
+			return nil, err
```

Comment:
Should we return if there is an error or just not set system.Type?

Reply:
I'm not following. Do you mean we should, instead of returning an error, return an `api.ResourcesSystem` just with `systemType` set?

Reply:
Before this PR, if the `sysfsExists` returned false then `GetSystem()` would return with `system.Type` set to the value from `systemGetType()` above.

Now with this change if something goes wrong in `getSystemFromLshw()` (for example if the command isn't available) then the `GetSystem()` function will return nothing but an error.

From what I've seen with the resources API in the past, the general pattern seems to be to try and acquire as much information as possible, but don't fail if certain parts are not available.

So this change seems to break with that approach.

Reply:
Changed.